source/common/upstream/cluster_manager_impl.cc

```diff
@@ -1488,7 +1489,14 @@ ClusterManagerImpl::ThreadLocalClusterManagerImpl::ClusterEntry::ClusterEntry(
                          parent.parent_.local_info_, parent.parent_, parent.parent_.runtime_,
                          parent.parent_.random_,
                          Router::ShadowWriterPtr{new Router::ShadowWriterImpl(parent.parent_)},
-                         parent_.parent_.http_context_, parent_.parent_.router_context_) {
+                         parent_.parent_.http_context_, parent_.parent_.router_context_),
+      quic_info_(
+#ifdef ENVOY_ENABLE_QUIC
+          Quic::createPersistentQuicInfoForCluster(parent.thread_local_dispatcher_, *cluster)
+#else
+          std::make_unique<Http::PersistentQuicInfo>()
```

Comment:
Is this required? I *think* `quic_info_()` should be equivalent. So if the else clause is removed I think it would be shorted, but effectively identical?

Reply:
It seems that `quic_info_()` won't instantiate an object but leave the unique_ptr to be nullptr.

Reply:
*facepalm* You're right of course. I misread this completely. Sorry!